common/src/main/java/org/conscrypt/SSLParametersImpl.java

```diff
@@ -202,14 +203,20 @@ X509TrustManager getX509TrustManager() {
      * @return the names of enabled cipher suites
      */
     String[] getEnabledCipherSuites() {
+        if (Arrays.asList(enabledProtocols).contains(NativeCrypto.SUPPORTED_PROTOCOL_TLSV1_3)) {
+            return SSLUtils.concat(
+                    NativeCrypto.SUPPORTED_TLS_1_3_CIPHER_SUITES, enabledCipherSuites);
+        }
         return enabledCipherSuites.clone();
     }
 
     /**
      * Sets the enabled cipher suites after filtering through OpenSSL.
      */
     void setEnabledCipherSuites(String[] cipherSuites) {
-        enabledCipherSuites = NativeCrypto.checkEnabledCipherSuites(cipherSuites).clone();
+        enabledCipherSuites = NativeCrypto.checkEnabledCipherSuites(
```

Comment:
Can you explain why we're filtering here? Is it because BoringSSL will reject it otherwise? Also, should we do this for the other versions of TLS as well?

Maybe add a comment to make it clear?

Reply:
Comment added.